### PR TITLE
Improve java debug performance to fetch list of threads/frames/variables

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Calculating debug location for every frame is quite expensive operation taking into account that user won't need it at all. It is batter to retrieve resource location on demand when user wants to open a resource in the editor. It took 2min to fetch list of threads and all frames for https://github.com/snowdrop/spring-boot-http-booster application, right now it takes only 1-2 sec

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/9243

### CODE REVIEW

**Comment by tolusha** [comment]
ci-test

**Review by ghost — APPROVED** [review]
100x berformance improvement! kudos to @tolusha 

I can confirm that the issue's resolved when spring boot is debugged in Che on OpenShift.

**Comment by gazarenkov** [comment]
WOW!

**Comment by garagatyi** [comment]
Yeah, 60x boost is impressive! 
Is it possible to have a PNG with both, so Stevan would be able to directly use them for a marketing purposes and we would be able to lurk too?

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/502/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/502//Selenium_tests_report)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10492)
docker image: eclipseche/che-server:10492
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by vparfonov — APPROVED** [review]

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/AbstractDebugger.java [hunk 1]
```diff
@@ -709,6 +709,15 @@ public void setValue(final Variable variable, final long threadId, final int fra
     }
   }
 
+  @Override
+  public Promise<? extends Location> getStackFrameLocation(long threadId, int frameIndex) {
+    if (isConnected()) {
+      return service.getStackFrameLocation(debugSessionDto.getId(), threadId, frameIndex);
+    }
+
+    return Promises.reject(JsPromiseError.create("Debugger is not connected"));
```
Comment (dkulieshov) [inline plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/AbstractDebugger.java]: If I'm not mistaken this method is deprecated and we should use `org.eclipse.che.api.promises.client.PromiseProvider#reject(org.eclipse.che.api.promises.client.PromiseError)`, no?

Comment (tolusha) [reply plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/AbstractDebugger.java]: @dkuleshov make sense

**Comment by tolusha** [comment]
ci-test

**Comment by SkorikSergey** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/506/) doesn't show any regression against this Pull Request.

**Pull request merged**